docs/how-to/react-server-components.md

```diff
@@ -1,50 +1,738 @@
 ---
-title: React Server Components
-# need to ship it first!
-hidden: true
+title: React Server Components (unstable)
+unstable: true
 ---
 
 # React Server Components
 
-<docs-info>This feature is still in development and not yet available.</docs-info>
+[MODES: data]
 
-In the future, async components can be rendered in loaders like any other data:
+<br/>
+<br/>
 
-```tsx filename=app/product-page.tsx
-// route("products/:pid", "./product-page.tsx");
-import type { Route } from "./+types/product";
-import Product from "./product";
-import Reviews from "./reviews";
+<docs-warning>React Server Components support is experimental and subject to breaking changes.</docs-warning>
 
-export async function loader({ params }: Route.LoaderArgs) {
-  return {
-    product: <Product id={params.pid} />,
-    reviews: <Reviews productId={params.pid} />,
-  };
+React Server Components (RSC) refers generally to an architecture and set of APIs provided by React since version 19.
+
+From the docs:
+
+> Server Components are a new type of Component that renders ahead of time, before bundling, in an environment separate from your client app or SSR server.
+> <cite>- [React "Server Components" docs][react-server-components-doc]</cite>
+
+React Router provides a set of APIs for integrating with RSC-native bundlers, allowing you to leverage [Server Components][react-server-components-doc] and [Server Functions][react-server-functions-doc] in your React Router applications.
```

Comment:
With Vite's RSC support currently being a plugin rather than being built-in, my original use of the word "native" doesn't quite apply.

```suggestion
React Router provides a set of APIs for integrating with RSC-compatible bundlers, allowing you to leverage [Server Components][react-server-components-doc] and [Server Functions][react-server-functions-doc] in your React Router applications.
```

Reply:
Ah yeah, that makes sense. Thanks for the suggested terminology update